praw/models/base.py

```diff
@@ -1,8 +1,9 @@
 """Provide the PRAWBase superclass."""
 from copy import deepcopy
-from typing import Any, Dict, Optional, TypeVar
+from typing import TYPE_CHECKING, Any, Dict, Optional
 
-Reddit = TypeVar("Reddit")
+if TYPE_CHECKING:  # pragma: no cover
+    from .. import Reddit
```

Comment:
For someone less familiar with `typing`, can you explain what this does? I assume that maybe the reference `"Reddit"` doesn't work unless the actual `Reddit` class exists. Is this correct?

Reply:
This is correct. But in 99% of cases, the import won't be executed unless a static type checker is running. I think it just needs to be present.